src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisDataFetcher.java

```diff
@@ -169,6 +193,11 @@ void advanceIteratorTo(String sequenceNumber, InitialPositionInStreamExtended in
             nextIterator = getIterator(ShardIteratorType.AT_SEQUENCE_NUMBER.toString(), sequenceNumber);
         }
         if (nextIterator == null) {
+            final Set<String> childShardIds = leaseCoordinator.getChildShardIds(shardId);
```

Comment:
In this case, I feel that there is no point to get the childShardIds from lease table because no matter whether childShards information exists in the lease table, we will always shutdown the shardConsumer with ShardEnd reason. The only point of checking childShardIds is to log a warn or error to indicate the missing information.